prusti-encoder/src/encoders/type/domain.rs

```diff
@@ -0,0 +1,599 @@
+        stronger_cons_axiom: bool,
+    ) -> DomainDataStruct<'vir> {
+        let name = self.domain.name();
+        let base = discr.map(|(_, _, v)| format!("{name}_{v}")).unwrap_or_else(|| name.to_string());
+        // Constructor
+        let field_snaps_to_snap = {
+            let name = vir::vir_format!(self.vcx, "{base}_cons");
+            self.functions.push(self.vcx.mk_domain_function(false, name, field_tys, self.self_ty));
+            FunctionIdent::new(name, vir::UnknownArity::new(field_tys))
+        };
+
+        // Variables and definitions useful for axioms
+        let fnames = (0..field_tys.len()).map(|idx| vir::vir_format!(self.vcx, "f{idx}")).collect::<Vec<_>>();
+        let cons_qvars: Vec<_> = field_tys.iter().enumerate().map(|(idx, ty)|
+            self.vcx.mk_local_decl(fnames[idx], ty)
+        ).collect();
+        let cons_qvars = self.vcx.alloc_slice(&cons_qvars);
+        let cons_args: Vec<_> = (0..field_tys.len()).map(|idx| self.vcx.mk_local_ex(fnames[idx])).collect();
+        let cons_call_with_qvars = if field_tys.is_empty() {
+            // TODO: workaround for `https://github.com/viperproject/silver/issues/236`
+            // remove once fixed.
+            field_snaps_to_snap.apply_ty(self.vcx, &cons_args, self.self_ty)
+        } else {
+            field_snaps_to_snap.apply(self.vcx, &cons_args)
+        };
+
+        // Discriminant axioms
+        if let Some((get_discr, val, _)) = discr {
+            let discr = get_discr.apply(self.vcx, [cons_call_with_qvars]);
+            let mut expr = self.vcx.mk_bin_op_expr(vir::BinOpKind::CmpEq, discr, val);
+            if !field_tys.is_empty() {
+                expr = self.vcx.mk_forall_expr(
+                    cons_qvars,
+                    self.vcx.alloc_slice(&[self.vcx.alloc_slice(&[discr])]),
+                    expr,
+                );
+            }
+            self.axioms.push(self.vcx.mk_domain_axiom(
+                vir::vir_format!(self.vcx, "ax_{base}_cons_discr"),
+                expr,
+            ));
+        }
+
+        // Accessors
+        let field_access = {
+            field_tys.iter().enumerate().map(|(idx, field_ty)| {
+                // Read
+                let name = vir::vir_format!(self.vcx, "{base}_read_{idx}");
+                let args = self.vcx.alloc_array(&[self.self_ty]);
+                let read = FunctionIdent::new(
+                    name,
+                    UnaryArity::new(args),
+                );
+                self.functions.push(self.vcx.mk_domain_function(false, name, args, field_ty));
+
+                let cons_read = read.apply(self.vcx, [cons_call_with_qvars]);
+                self.axioms.push(self.vcx.mk_domain_axiom(
+                    vir::vir_format!(self.vcx, "ax_{base}_cons_read_{idx}"),
+                    self.vcx.mk_forall_expr(
+                        cons_qvars,
+                        self.vcx.alloc_slice(&[self.vcx.alloc_slice(&[cons_read])]),
```

Comment:
I believe the trigger here should be `cons_call_with_qvars` instead of `cons_read`. That would correspond to #20. Unless you have found out that this is no longer a desirable change?

As it currently is 

```rust
extern crate prusti_contracts;
use prusti_contracts::*;

#[requires(b)]
fn assert_true(b: bool) {}

fn main() {
    assert_true( 1 + 1 > 1);
}
```
does not verify but with the different trigger it does.